frontend/components/details/ModuleHTML.tsx

```diff
@@ -0,0 +1,60 @@
+import { HTMLReactParserOptions } from "html-react-parser";
+import { Element } from "html-react-parser/node_modules/domhandler/lib/node";
+import { FC, ReactElement, useEffect, useState } from "react";
+
+interface ModuleHTMLProps {
+  html: string;
+}
+
+const ModuleHTML: FC<ModuleHTMLProps> = ({ html }: ModuleHTMLProps) => {
+  const [renderedHTML, setRenderedHTML] = useState(() => (
+    <div dangerouslySetInnerHTML={{ __html: html }} />
+  ));
```

Comment:
Using state here allows faster initial render times (especially useful for SSR) since all the expensive logic is behind `useEffect`.